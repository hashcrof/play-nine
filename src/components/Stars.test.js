import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Stars from './Stars'

const generateRandomNumberOfStars = () => {
  return 1 + Math.floor(Math.random() * 9)
}

it('renders the Stars component', () =>{
  const output = shallow(
    <Stars numberOfStars={3} />
  )
  expect(shallowToJson(output)).toMatchSnapshot()
});

it('has a certain number of stars', () =>{
  const numberOfStars = generateRandomNumberOfStars()
  const wrapper = mount(
    <Stars numberOfStars={numberOfStars} />
  )

  const div = wrapper.find('div.stars')
  expect(div.props().children).toHaveLength(numberOfStars)
});
