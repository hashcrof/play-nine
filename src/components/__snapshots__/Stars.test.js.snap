// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Stars component 1`] = `
<div
  className="col-5 stars"
>
  <i
    className="fa fa-star"
    key="0"
  />
  <i
    className="fa fa-star"
    key="1"
  />
  <i
    className="fa fa-star"
    key="2"
  />
</div>
`;
